### PR TITLE
[DownloadBuildArtifactsV0] Respect "Parallelization limit" input

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Task name**: 
- DownloadBuildArtifactsV0

**Description**: 
This PR provide a fix for the issue when the task didn't respect the `Parallelization limit` input.

_Changes_:
- Added `resolveParallelProcessingLimit` function to set `Parallelization limit` value
- Changed `configureDownloaderOptions` function to respect `Parallelization limit` input

_Other changes:_
- Removed `timeoutPromise` function since not used in task

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:**
- [#873](https://github.com/microsoft/build-task-team/issues/873)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by alexander-smolyakov** [comment]
Test pipeline: [link](https://dev.azure.com/v-alsmo/ArtifactsTest/_build/results?buildId=1066&view=results)

File: Tasks/DownloadBuildArtifactsV0/download_helper.ts [hunk 1]
```diff
@@ -96,3 +87,47 @@ function isItemCorrupted(ticket: ArtifactDownloadTicket): boolean {
...
+ * @returns {number} - parallel processing limit
+ */
+export function resolveParallelProcessingLimit(artifactDownloadLimit: string, taskLimit: string): number {
+    const artifactDownloadParallelLimit: number = Number(artifactDownloadLimit);
+    if (!isNaN(artifactDownloadParallelLimit) && artifactDownloadParallelLimit > 0) {
+        debug('The value from "release.artifact.download.parallellimit" variable will be used in the task');
+        debug(`The parallelization limit is set to ${artifactDownloadParallelLimit}`);
+        return artifactDownloadParallelLimit;
+    } else {
+        debug('The "release.artifact.download.parallellimit" variable contains the incorrect value');
```
Comment (anatolybolshakov) [inline Tasks/DownloadBuildArtifactsV0/download_helper.ts:118]: Could it make sense to print value here for visibility?

File: Tasks/DownloadBuildArtifactsV0/download_helper.ts [hunk 2]
```diff
@@ -96,3 +87,47 @@ function isItemCorrupted(ticket: ArtifactDownloadTicket): boolean {
...
+        debug('The value from "Parallelization limit" input will be used in the task');
+        debug(`The parallelization limit is set to ${taskInputParallelLimit}`);
+        return taskInputParallelLimit;
+    } else {
+        debug('The "Parallelization limit" input contains the incorrect value');
+    }
+
+    debug('The parallelization limit is set to default value - 8');
+
+    return 8;
```
Comment (anatolybolshakov) [inline Tasks/DownloadBuildArtifactsV0/download_helper.ts:132]: Could you please move it to constants? Could it make sense to reuse it in a message above via templates?

File: Tasks/DownloadBuildArtifactsV0/download_helper.ts [hunk 3]
```diff
@@ -96,3 +87,47 @@ function isItemCorrupted(ticket: ArtifactDownloadTicket): boolean {
...
+ *
+ * If validation failed for both cases the function will return the default value for the `parallelProcessingLimit` option.
+ *
+ * @param artifactDownloadLimit {string} - value of `release.artifact.download.parallellimit` variable
+ * @param taskLimit {string} - value of `Parallelization limit` task input
+ * @returns {number} - parallel processing limit
+ */
+export function resolveParallelProcessingLimit(artifactDownloadLimit: string, taskLimit: string): number {
+    const artifactDownloadParallelLimit: number = Number(artifactDownloadLimit);
+    if (!isNaN(artifactDownloadParallelLimit) && artifactDownloadParallelLimit > 0) {
```
Comment (anatolybolshakov) [inline Tasks/DownloadBuildArtifactsV0/download_helper.ts:113]: Nit: looks like some duplication here, could it make sense to move validation somewhere like separate function?

Comment (alexander-smolyakov) [reply Tasks/DownloadBuildArtifactsV0/download_helper.ts]: Good point. I extracted check logic into separate function.

File: Tasks/DownloadBuildArtifactsV0/download_helper.ts (hunk 2 above)
Comment (alexander-smolyakov) [reply Tasks/DownloadBuildArtifactsV0/download_helper.ts]: I moved this value to constants thanks. Thanks for the point!

File: Tasks/DownloadBuildArtifactsV0/download_helper.ts (hunk 1 above)
Comment (alexander-smolyakov) [reply Tasks/DownloadBuildArtifactsV0/download_helper.ts]: I reworked debug messages, thanks!

**Review by anatolybolshakov — APPROVED** [review]

**Comment by anatolybolshakov** [comment]
Tested on basic scenario also -  'Parallelization limit' input is respected for me now.

**Review by EzzhevNikita — APPROVED** [review]
LGTM! Tested this changes manually, _resolveParallelProcessingLimit_ logic work good for me. Test [pipeline](https://dev.azure.com/v-niezz-testorg/TestProject/_build/results?buildId=1949&view=results)

**Pull request merged**